ZenPacks/zenoss/Microsoft/Windows/tests/testServicesRegex.py

```diff
@@ -96,3 +107,52 @@ def test_model_service_multiply_values(self):
             if not omit_service(model_list, ignore_list, service, self.log):
                 result.append(service)
         self.assertEqual(sorted(result), sorted(expected))
+
+    def test_model_service_grouped(self):
+        self.device.zWinServicesNotModeled = []
+        self.device.zWinServicesModeled = []
+        self.device.zWinServicesGroupedByClass = ['CDPUserSvc']
+        _, _, group_list = validate_modeling_regex(self.device, self.log)
+        service_class_name, service_class_caption = truncate_service_class(group_list, 'CDPUserSvc_123asdl', 'Contact Data_123asdl')
+        self.assertEqual(service_class_name, 'CDPUserSvc')
+        self.assertEqual(service_class_caption, 'Contact Data')
+
+    def test_model_service_grouped_empty(self):
+        self.device.zWinServicesNotModeled = []
+        self.device.zWinServicesModeled = []
+        self.device.zWinServicesGroupedByClass = []
+        _, _, group_list = validate_modeling_regex(self.device, self.log)
+        service_class_name, service_class_caption = truncate_service_class(group_list, 'PimIndexMaintenanceSvc_aeff1234', 'Contact Data_aeff1234')
+        self.assertEqual(service_class_name, 'PimIndexMaintenanceSvc_aeff1234')
+        self.assertEqual(service_class_caption, 'Contact Data_aeff1234')
+
+    def test_model_service_grouped_multiple(self):
```

Comment:
might want to include tests for grouping services with multiple underscores in the name